### DISCUSSION TITLE
L1, L2, L4 pixels

### DISCUSSION THREAD

**Discussion by GihanSoft** [SixLabors/ImageSharp#2036] [Ideas] [answered]
in PDF files, PNG images saves with bit depth of 1, 2, 4 and 8.
I need to load pixel data from it. 
For 8 there is L8 but not for others.

**Comment by JimBobSquarePants** [comment, marked as answer]
You're mixing up what bit depth represents in PNG. 

PNG bit depth of 1 means 1 bit per sample or per palette index (not per pixel).

A 1 bit PNG will contain up to 2 colors in the palette.
A 2 bit PNG will contain up to 4 colors in the palette.
_and so on..._

Those colors are stored in RGBA format (32bit).

L8 is a 1 byte (8 bits) pixel format that is generally used to describe luminance (grayscale). You would use this if you new that your image was grayscale only or you wanted it to be.

**Reply by GihanSoft** [reply]
but image size is actually like it take 1 bit per pixel. 
`with` * `height` == `pixelData` * 8
(not exactly equal but it was near)
(`pixedData` is from PDF.)

**Reply by GihanSoft** [reply]
I also encoded an empty PNG with same size, same depth (1bit) and no compression and size was near pixel data from PDF

**Reply by JimBobSquarePants** [reply]
I recommend reading the [PNG specification](https://www.w3.org/TR/PNG/). There are hundreds of factors that can affect the output size of an encoded PNG image.
